vortex-cxx/examples/hello-vortex.cpp

```diff
@@ -13,8 +13,16 @@ int main(int argc, char *argv[]) {
         return 1;
     }
     std::string vortex_file = argv[1];
-    auto file = vortex::VortexFile::Open(vortex_file);
-    auto stream = file.CreateScanBuilder().WithLimit(100).IntoStream();
+    // 1. bad, dangling lvalue reference!
+    // auto& builder = vortex::VortexFile::Open(vortex_file).CreateScanBuilder().WithLimit(3);
+    // auto stream = std::move(builder).IntoStream();
+    // 2. good, prefer
+    auto builder = vortex::VortexFile::Open(vortex_file).CreateScanBuilder();
```

Comment:
Any combination of this should work without subtle errors.

Reply:
Great. I modified the examples.